app/main.py

```diff
@@ -20,6 +28,20 @@
 def read_root(request: Request):
     return templates.TemplateResponse("index.html", {"request": request})
 
+
+def generate_headline(text: str, image_url: str) -> str:
```

Comment:
Put this in a separate file.

Reply:
Sure, I'll move the `generate_headline` function to a separate file and import it into `app/main.py`.

### Steps to Address the Comment

1. Create a new file `app/openai_utils.py`.
2. Move the `generate_headline` function to `app/openai_utils.py`.
3. Import the `generate_headline` function in `app/main.py`.

### Updated Code

#### Create `app/openai_utils.py`

```python
import openai

def generate_headline(text: str, image_url: str) -> str:
    prompt = f"Generate an ad headline for the following image URL based on the website text:\n\nWebsite Text:\n{text}\n\nImage URL: {image_url}\n\nAd Headline:"
    response = openai.ChatCompletion.create(
        model="gpt-3.5-turbo",
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": prompt}
        ],
        max_tokens=50
    )
    return response.choices[0].message['content'].strip()
```

#### Update `app/main.py`

```python
from fastapi import FastAPI, HTTPException, Query, Request
from fastapi.responses import HTMLResponse
from fastapi.staticfiles import StaticFiles
from fastapi.templating import Jinja2Templates
from pydantic import HttpUrl
import requests
from bs4 import BeautifulSoup
import os
from app.scraping import extract_images
from app.openai_utils import generate_headline  # Import the function here
from dotenv import load_dotenv
import openai

# Load environment variables
load_dotenv()

# Initialize OpenAI API key
openai.api_key = os.getenv("OPENAI_API_KEY")

app = FastAPI()

# Mount the static files directory
app.mount("/static", StaticFiles(directory="templates/static"), name="static")

# Configure Jinja2 templates
templates = Jinja2Templates(directory="templates")

@app.get("/", response_class=HTMLResponse)
def read_root(request: Request):
    return templates.TemplateResponse("index.html", {"request": request})

@app.get("/extract-text")
def extract_text(url: HttpUrl = Query(..., description="The URL to extract text and images from")):
    headers = {
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3",
        "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*;q=0.8",
    }
    try:
        response = requests.get(url, headers=headers)
        response.raise_for_status()
    except requests.RequestException as e:
        raise HTTPException(status_code=400, detail=str(e))

    soup = BeautifulSoup(response.content, "html.parser")
    text = soup.get_text(separator="\n", strip=True)
    text = "\n".join([line for line in text.split("\n") if line.strip()])  # Remove empty lines

    # Extract images using the utility method
    images = extract_images(soup)

    # Generate headlines for each image
    headlines = [generate_headline(text, image) for image in images]

    return {"text": text, "images": images, "headlines": headlines}
```

I'll go ahead and make these changes in the PR.